service/src/main/resources/templates/index.html

```diff
@@ -110,7 +85,48 @@
     };
   }
 
-  window.onload = function() {
+    // Examples cache
+    const examplesCache = {};
+
+    // Custom plugin for logic that happens before a parameters element is created, hardcoded
+    // to work with the dataset example in the PUT dataset operation.
+    const LoadExternalValuePlugin = () => {
+      return {
+        wrapComponents: {
+          // wrapping https://github.com/swagger-api/swagger-ui/blob/master/src/core/components/parameters/parameters.jsx
+          parameters: (Original, { React }) => (props) => {
+            // These hardcoded values should be replaced with code that determines the values
+            // using the user's current selection.
+            const contentType = 'application/json'
+            const activeExamplesKey = 'dataset'
+            const operation = props.operation
+            const externalValue = operation.getIn(['requestBody', 'content', contentType, 'examples', activeExamplesKey, 'externalValue'])
+
+            if (externalValue) {
+              if (examplesCache[externalValue]) {
+                // If the examples cache contains the externalValue key, set the example value
+                // directly from the cache.
+                props.operation = operation.setIn(['requestBody', 'content', contentType, 'examples', activeExamplesKey, 'value'], examplesCache[externalValue])
+              } else {
+                // Fetch the external file.
+                fetch(externalValue)
+                  .then(res => res.text())
+                  .then(data => {
+                    // Put fetched file content into the cache.
+                    examplesCache[externalValue] = JSON.parse(data)
+                    // At this point the cache contains the JSON data but the UI is not refreshed.
+                    // The code shold be able to use `oas3Actions` to refresh the UI.
+                  })
+                  .catch(e => console.error(e))
+              }
+            }
+            return React.createElement(Original, props)
+          }
+        }
+      }
+    }
```

Comment:
I put this together, and it seems to work for updating the example. I believe it is a bit more in model for react code. If you would prefer to revisit in a different PR feel free to reject the suggestion.


```suggestion
    const LoadExternalValuePlugin = () => {
      return {
        wrapComponents: {
          // wrapping https://github.com/swagger-api/swagger-ui/blob/master/src/core/components/parameters/parameters.jsx
          parameters: (Original, { React, oas3Actions }) => (props) => {
            const [example, setExample] = React.useState()
            // These hardcoded values should be replaced with code that determines the values
            // using the user's current selection.
            const {operation, pathMethod} = props
            const contentType = 'application/json'
            const activeExamplesKey = 'dataset'
            const externalValue = operation.getIn(['requestBody', 'content', contentType, 'examples', activeExamplesKey, 'externalValue'])

            // Since openAPI3 (oas3) is responsible for rendering, use a side effect to force a re-render via the oas3 UI API
            // Normally, in react the setState would be enough to get the rendered state we are after
            React.useEffect(() => {
              oas3Actions.setRequestBodyValue({ pathMethod, value: example })
            }, [example])

            if (externalValue && !example) {
                fetch(externalValue)
                  .then(res => res.text())
                  .then(data => setExample(JSON.parse(data)))
                  .catch(e => console.error(e))
            }

            // Since we need to modify the props for everything to work correctly, let's be explicit about this
            const updatedProps = example ? {
                ...props,
                ...{operation: operation.setIn(['requestBody', 'content', contentType, 'examples', activeExamplesKey, 'value'], example)}
              } : props
            return React.createElement(Original, updatedProps)
          }
        }
      }
    }

```

Reply:
Thanks, I'll give this try

Reply:
This worked to update the example, although the state still isn't quite correct (the example select menu isn't in the correct state) until the Reset button is pressed. I tried writing code to press the button automatically but couldn't get it to work.

I changed things around a little in your code, let me know what you think. I wanted to check `externalValue` separately to make it clear that the code path is only taken if `externalValue` is present. I also wanted to make the scope of `example` and `setExample` as small as possible.